rasa_sdk/tracing/instrumentation/attribute_extractors.py

```diff
@@ -56,3 +56,32 @@ def extract_attrs_for_validation_action(
         "slots_to_validate": json.dumps(list(slots_to_validate)),
         "action_name": self.name(),
     }
+
+
+def extract_attrs_for_action_executor_create_api_response(
+    events: List[Dict[Text, Any]],
+    messages: List[Dict[Text, Any]],
+) -> Dict[Text, Any]:
+    """Extract the attributes for `ActionExecutor.run`.
+
+    :param events: A list of events.
+    :param messsages: A list of bot responses.
+    :return: A dictionary containing the attributes.
+    """
+    event_names = []
+    slot_names = []
+
+    for event in events:
+        event_names.append(event.get("event"))
+        if event.get("event") == "slot" and event.get("name") != "requested_slot":
+            slot_names.append(event.get("name"))
+    utters = [
+        message.get("response") for message in messages if message.get("response")
+    ]
+
+    return {
+        "events": json.dumps(list(dict.fromkeys(event_names))),
+        "slots": json.dumps(list(dict.fromkeys(slot_names))),
```

Comment:
Wouldn't it be more readable to pass directly `event_names` or `slot_names` to `json.dumps`? these vars are already lists of strings 🤔

Reply:
I had to remove duplicates without making it a set.

Reply:
Ah got it 👍🏻